Support build multi boards with Yocto Styhead

khanhduydo-rvc opened the pull request.

tiennguyenuh commented on include/core-image-renesas-base.inc:
```diff
@@ -213,8 +212,3 @@ EXTRA_IMAGEDEPENDS += " host-readme "
-
-# Readme documentation
-EXTRA_IMAGEDEPENDS += " rzg2l-sbc-readme "
-EXTRA_IMAGEDEPENDS += " rzg2l-sbc-docs "
```
Please keep these lines, but change rzg2l-sbc to rz-cmn and update the folder/recipes name accordingly.

tiennguyenuh commented on include/core-image-renesas-qt.inc:
Please keep this file, we might need it for renesas-weston and quickboot images.

tiennguyenuh commented on include/core-image-renesas-qt6.inc:
Please keep this file, we might need it for renesas-weston and quickboot images.

tiennguyenuh commented on include/core-image-renesas-quickboot.inc:
Please keep this file, we might need it for quickboot images.

tiennguyenuh commented on recipes-bsp/firmware-pack/firmware-pack.bb:
```diff
@@ -7,45 +7,64 @@ inherit deploy
+
+            # Create bl2_bp.bin mmc
+	        bptool ${RECIPE_SYSROOT}/boot/bl2-${target}.bin bp.bin 0x08103000 mmc
+	        cat bp.bin ${RECIPE_SYSROOT}/boot/bl2-${target}.bin > bl2_bp_mmc_${target}.bin
```
Check alignment.

tiennguyenuh commented on recipes-bsp/firmware-pack/bptool-native.bb:
```diff
@@ -0,0 +1,28 @@
+SECTION = "bootloaders"
+SUMMARY = "Bptool from Trusted Firmware-A"
```
Please add a description to clarify that this tool is intended for use with the RZ/V2H platform only, as it's not a generic tool for all Renesas SoCs.

tiennguyenuh commented on recipes-bsp/firmware-pack/bptool-native.bb:
```diff
@@ -0,0 +1,28 @@
+
+LIC_FILES_CHKSUM = "file://docs/license.rst;md5=b2c740efedc159745b9b31f88ff03dde"
+
+URL = "git://github.com/vudangRVC/rz-atf-sst.git"
```
Just a reminder, we will integrate RZ/V2H into Renesas-SST and update this URL to point to the SST repository.

tiennguyenuh commented on recipes-bsp/flash-writer/flash-writer.bb:
```diff
@@ -11,61 +11,55 @@ ENABLE_RZ_SCE		?= '0'
+	for target in ${TARGETS}; do
+		PMIC_BUILD_DIR="${S}/${target}/build_pmic"
+		install -m 644 ${S}/${target}/AArch64_output/*.mot ${DEPLOYDIR}/target/images
+			if [ "${PMIC_SUPPORT}" = "1" ]; then
+				install -m 644 ${PMIC_BUILD_DIR}/*.mot ${DEPLOYDIR}/target/images
```
Check alignment.

khanhduydo-rvc replied:
i don't see any alignment issue here

tiennguyenuh commented on recipes-kernel/linux/linux-yocto_6.10.bbappend:
```diff
@@ -1,24 +1,23 @@
-COMPATIBLE_MACHINE:rzg2l-sbc = "(rzg2l-sbc)"
-COMPATIBLE_MACHINE = "^(aarch64|rzg2l-sbc)$"
+COMPATIBLE_MACHINE:rz-cmn = "(rz-cmn)"
+COMPATIBLE_MACHINE = "^(aarch64|rzg2l-sbc|rz-cmn)$"
```
Remove rzg2l-sbc from COMPATIBLE_MACHINE, as it has been replaced by rz-cmn

tiennguyenuh commented on recipes-kernel/linux/linux-yocto_6.10.bbappend:
```diff
@@ -41,21 +40,23 @@ SRC_URI:append:rzg2l-sbc = "\
+                    file://multi-boards-support/0001-rzv2l-evk-add-wm8978-codec-to-support-audio-in-v2l-e.patch \
+                    file://multi-boards-support/0002-rzv2h-evk-Support-RZV2H-EVK.patch \
+"
+
```
Can you organize the folder structure as follows for better management?

```
rz-cmn/
├── cfgs
└── patches
    ├── common
    ├── rzg2l-sbc
    ├── rzv2h-evk
    └── rzv2l-evk
```

tiennguyenuh commented on recipes-kernel/linux/linux-yocto_6.10.bbappend:
```diff
@@ -41,21 +40,23 @@ SRC_URI:append:rzg2l-sbc = "\
                     file://dts-patches/0009-rzg2l-sbc-Support-more-fourcc-format-for-CRU.patch \
                 "
```
These patches should belong to 'rzg2l-sbc'. Please move them to the 'rzg2l-sbc' folder and update the recipe accordingly.

khanhduydo-rvc replied:
ok

tiennguyenuh commented on recipes-extend/host-readme/files/Readme.md:
```diff
@@ -1,33 +1,38 @@
-│   ├── rzsbc_yocto.sh
-│   └── site.conf
-└── tools/                                                                                      # Tools and scripts used for managing and flashing bootloaders and filesystems across different platforms
+│   ├── core-image-bsp-rz-cmn-20250523093711.rootfs.manifest         -> Manifest file for the root filesystem
```
Update the hierarchy for target images, as we will no longer retain the machine in the target.
Please make it consistent with the other README files.

tiennguyenuh commented on include/core-image-renesas-qt6.inc:
Please keep this file, as we might need to use it for renesasweston and quickboot images.

tiennguyenuh commented on recipes-bsp/flash-writer/flash-writer.bb:
```diff
@@ -11,61 +11,61 @@ ENABLE_RZ_SCE		?= '0'
+UNPACKDIR = "${S}"
 
-	oe_runmake BOARD=${BOARD}
+TARGETS = "rzg2l-sbc rzg2l-evk rzv2l-evk"
```
Find a better approach to define supported machine for multi-platform

khanhduydo-rvc committed:
rz-cmn: Modify recipes-kernel folder

- Add build support for rz-cmn MACHINE
- Modify linux-yocto recipes to support 4 boards: rzg2l-sbc, rzg2l-evk, rzv2l-evk, rzv2h-evk

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Modify recipes-bsp folder

- Add build support for flash writer, ATF, and U-Boot for 4 boards: rzg2l-sbc, rzg2l-evk, rzv2l-evk, rzv2h-evk

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

khanhduydo-rvc committed:
rz-cmn: Modify remaining folders

- Add build support for rz-cmn MACHINE
- Modify Readme to adapt to new MACHINE

Signed-off-by: Khanh Do Duy <khanh.do.yg@bp.renesas.com>

tiennguyenuh commented on recipes-kernel/linux/rz-cmn/rzg2l-sbc/panfrost.cfg:
The `panfrost.cfg` file should be used commonly/shared for rzg2l series. It should not be placed in the `rzg2l-sbc` folder
